desc/objectives/_stability.py

```diff
@@ -367,3 +369,210 @@ def compute(self, params, constants=None):
+        len_grid = LinearGrid(rho=1.0, M=eq.M_grid, N=eq.N_grid, NFP=eq.NFP)
+        self._len_keys = ["a"]
+        len_profiles = get_profiles(self._len_keys, obj=eq, grid=len_grid)
+        len_transforms = get_transforms(self._len_keys, obj=eq, grid=len_grid)
+
+        # make a set of nodes along a single fieldline
+        zeta = np.linspace(-self.zetamax, self.zetamax, self.nzeta)
+
+        self._dim_f = 1
+        self._data_keys = ["ideal ball gamma2"]
+
+        self._args = get_params(
+            self._iota_keys + self._len_keys + self._data_keys,
+            obj="desc.equilibrium.equilibrium.Equilibrium",
+            has_axis=False,
+        )
+
+        self._constants = {
+            "iota_transforms": iota_transforms,
+            "iota_profiles": iota_profiles,
+            "len_transforms": len_transforms,
+            "len_profiles": len_profiles,
+            "rho": self.rho,
+            "alpha": self.alpha,
+            "zeta": zeta,
+            "quad_weights": 1.0,
+        }
+        super().build(use_jit=use_jit, verbose=verbose)
+
+    def compute(self, params, constants=None):
+        """
+        Compute the ballooning stability growth rate.
+
+        Parameters
+        ----------
+        params : dict
+            Dictionary of equilibrium degrees of freedom, eg Equilibrium.params_dict
+        constants : dict
+            Dictionary of constant data, eg transforms, profiles etc. Defaults to
+            self.constants
+
+        Returns
+        -------
+        gamma : ndarray
+            ideal ballooning growth rate.
+
+        """
+        eq = self.things[0]
+
+        if constants is None:
+            constants = self.constants
+        # we first compute iota on a uniform grid to get correct averaging etc.
+        iota_data = compute_fun(
+            "desc.equilibrium.equilibrium.Equilibrium",
+            self._iota_keys,
+            params=params,
+            transforms=constants["iota_transforms"],
+            profiles=constants["iota_profiles"],
+        )
+
+        len_data = compute_fun(
+            "desc.equilibrium.equilibrium.Equilibrium",
+            self._len_keys,
+            params=params,
+            transforms=constants["len_transforms"],
+            profiles=constants["len_profiles"],
+        )
+
+        # Now we compute theta_DESC for given theta_PEST
+        rho, alpha, zeta = constants["rho"], constants["alpha"], constants["zeta"]
+
+        # we prime the data dict with the correct iota values so we don't recompute them
+        # using the wrong grid
+        data = {
+            "iota": iota_data["iota"][0],
+            "iota_r": iota_data["iota_r"][0],
+            "shear": iota_data["shear"][0],
+            "a": len_data["a"],
+        }
+
+        grid = eq.get_rtz_grid(
+            rho,
+            alpha,
+            zeta,
+            coordinates="raz",
+            period=(np.inf, 2 * np.pi, np.inf),
+        )
```

Comment:
You should pass in the `params` here like `ripple` and `Gamma_c` do
```
        grid = eq.get_rtz_grid(
...
            params=params,
        )
```
so that the coordinate mapping uses the new values for spectral coefficients for the poloidal stream function after they are perturbed. Right now this uses the values of the initial equilibrium.

Reply:
Done.